modules/io/src/image/vpImageIo.cpp

```diff
@@ -558,1681 +349,250 @@ void vpImageIo::write(const vpImage<vpRGBa> &I, const std::string &filename)
   }
 
   if (try_opencv_writer) {
-#if VISP_HAVE_OPENCV_VERSION >= 0x020100
-    // std::cout << "Use opencv to write the image" << std::endl;
-    cv::Mat cvI;
-    vpImageConvert::convert(I, cvI);
-    cv::imwrite(filename, cvI);
+#if defined(VISP_HAVE_OPENCV) && VISP_HAVE_OPENCV_VERSION >= 0x020100
+    writeOpenCV(I, filename, 90);
 #else
-    switch (getFormat(filename)) {
-    case FORMAT_JPEG:
-      writeJPEG(I, filename);
-      break;
-    case FORMAT_PNG:
-      writePNG(I, filename);
-      break;
-    case FORMAT_BMP:
-    case FORMAT_TIFF:
-    case FORMAT_DIB:
-    case FORMAT_PBM:
-    case FORMAT_RASTER:
-    case FORMAT_JPEG2000:
-    case FORMAT_UNKNOWN:
-    default:
-      vpCERROR << "Cannot write file: Image format not supported..." << std::endl;
-      throw(vpImageException(vpImageException::ioError, "Cannot write file: Image format not supported"));
-  }
+    std::string message = "Cannot write file \"" + filename + "\": No backend able to support this image format";
+    throw(vpImageException(vpImageException::ioError, message));
 #endif
   }
 }
 
-//--------------------------------------------------------------------------
-// PFM
-//--------------------------------------------------------------------------
-
-/*!
-  Write the content of the image bitmap in the file which name is given by \e
-  filename. This function is built like portable gray pixmap (eg PGM P5) file.
-  but considers float image data.
-
-  \param I : Image to save as a (PFM P8) file.
-  \param filename : Name of the file containing the image.
-*/
-
-void vpImageIo::writePFM(const vpImage<float> &I, const std::string &filename)
+void vpImageIo::readJPEG(vpImage<unsigned char> &I, const std::string &filename, int backend)
```

Comment:
Should be updated to not throw an exception if the backend is not available but instead print an error and use either `stb` or `simd` as they are always available depending on the performance numbers.

Same for PNG.